sdk/src/Core/Amazon.Runtime/Credentials/Internal/AwsV4aAuthScheme.cs

```diff
@@ -31,5 +33,13 @@ public IIdentityResolver GetIdentityResolver(IIdentityResolverConfiguration conf
         {
             return configuration.GetIdentityResolver<AWSCredentials>();
         }
+
+        /// <inheritdoc/>
+        public ISigner Signer()
+        {
+            if (_signer == null)
+                _signer = new AWS4aSignerCRTWrapper();
```

Comment:
To avoid weird threading issues can we use the Interlock class for this lazy setting.

```
Interlocked.Exchange(ref _signer, new AWS4aSignerCRTWrapper());
```

Reply:
Updated.